From 25bf677f5cc6d279abd2ac314b56291abee0e994 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Wed, 20 Nov 2019 17:01:54 +0200
Subject: [PATCH] Update xen_back DMA tweaks

According to recent Kingfisher related changes to Linux we have
to update "of_dma_configure" API usage and remove "force_dma" flag.

Please note, this patch will be only applied when building
h3ulcb-4x2g-kf machine.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 services/system/common/env/xen/xen_back.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/services/system/common/env/xen/xen_back.c b/services/system/common/env/xen/xen_back.c
index 187738a..06072b2 100644
--- a/services/system/common/env/xen/xen_back.c
+++ b/services/system/common/env/xen/xen_back.c
@@ -661,9 +661,8 @@ static int xdrv_probe(struct xenbus_device *xb_dev,
 	 * is not correct: to fix this call of_dma_configure() with a NULL
 	 * node to set default DMA ops.
 	 */
-	xb_dev->dev.bus->force_dma = true;
 	xb_dev->dev.coherent_dma_mask = DMA_BIT_MASK(40);
-	of_dma_configure(&xb_dev->dev, NULL);
+	of_dma_configure(&xb_dev->dev, NULL, true);
 #endif
 	drv_info->dev_id = VGSX_BAD_DEVICE_ID;
 	INIT_WORK(&drv_info->op_work, xdrv_do_op);
-- 
2.7.4

